packages/contract-wrappers/src/contract_wrappers/coordinator_wrapper.ts

```diff
@@ -0,0 +1,672 @@
+        const makerAddress = getMakerAddressOrThrow(orders);
+        assert.isETHAddressHex('feeRecipientAddress', feeRecipientAddress);
+        assert.isSenderAddressAsync('makerAddress', makerAddress, this._web3Wrapper);
+
+        const data = this._transactionEncoder.batchCancelOrdersTx(orders);
+        const transaction = await this._toSignedZeroExTransactionAsync(data, makerAddress);
+        return (await this._requestServerResponseAsync(
+            transaction,
+            makerAddress,
+            feeRecipientAddress,
+        )) as CoordinatorServerCancellationResponse;
+    }
+
+    /**
+     * Hard cancellation on Exchange contract. Cancels a single order.
+     * @param   order           An object that conforms to the Order or SignedOrder interface. The order you would like to cancel.
+     * @param   orderTransactionOpts Optional arguments this method accepts.
+     * @return  Transaction hash.
+     */
+    @decorators.asyncZeroExErrorHandler
+    public async hardCancelOrderAsync(
+        order: Order | SignedOrder,
+        orderTransactionOpts: OrderTransactionOpts = { shouldValidate: true },
+    ): Promise<string> {
+        assert.doesConformToSchema('order', order, schemas.orderSchema);
+        assert.doesConformToSchema('orderTransactionOpts', orderTransactionOpts, orderTxOptsSchema, [txOptsSchema]);
+        await assert.isSenderAddressAsync('makerAddress', order.makerAddress, this._web3Wrapper);
+
+        const data = this._transactionEncoder.cancelOrderTx(order);
+        const transaction = await this._toSignedZeroExTransactionAsync(data, order.makerAddress);
+        return this._sendTransactionAsync(
+            transaction,
+            order.makerAddress,
+            transaction.signature,
+            [],
+            [],
+            orderTransactionOpts,
+        );
+    }
+
+    /**
+     * Hard cancellation on Exchange contract.
+     * Batch version of cancelOrderAsync. Executes multiple cancels atomically in a single transaction.
+     * @param   orders                An array of orders to cancel.
+     * @param   orderTransactionOpts  Optional arguments this method accepts.
+     * @return  Transaction hash.
+     */
+    @decorators.asyncZeroExErrorHandler
+    public async batchHardCancelOrdersAsync(
+        orders: SignedOrder[],
+        orderTransactionOpts: OrderTransactionOpts = { shouldValidate: true },
+    ): Promise<string> {
+        assert.doesConformToSchema('orders', orders, schemas.ordersSchema);
+        const makerAddress = getMakerAddressOrThrow(orders);
+        assert.doesConformToSchema('orderTransactionOpts', orderTransactionOpts, orderTxOptsSchema, [txOptsSchema]);
+        await assert.isSenderAddressAsync('makerAddress', makerAddress, this._web3Wrapper);
+
+        const data = this._transactionEncoder.batchCancelOrdersTx(orders);
+        const transaction = await this._toSignedZeroExTransactionAsync(data, makerAddress);
```

Comment:
Do we need to do it this way for hard cancels? The resulting UX would be 2 MM prompts wouldn't it?

Reply:
Not sure I understand - is that because the Exchange method also does a MM prompt? 

Looks like the coordinator contract under the hood skips everything for cancels and forwards directly to the Exchange contract so we could call the exchange method directly.

Reply:
Ignore this my brain wasn't working!